src/helpers/component-tree.ts

```diff
@@ -0,0 +1,89 @@
+import { ReactTestInstance } from 'react-test-renderer';
+
+/**
+ * Checks if the given element is a host element.
+ * @param element The element to check.
+ */
+export function isHostElement(element?: ReactTestInstance | null): boolean {
+  return typeof element?.type === 'string';
+}
+
+/**
+ * Returns first host ancestor for given element.
+ * @param element The element start traversing from.
+ */
+export function getHostParent(
+  element: ReactTestInstance | null
+): ReactTestInstance | null {
+  if (element == null) {
+    return null;
+  }
+
+  let current = element.parent;
+  while (current) {
+    if (isHostElement(current)) {
+      return current;
+    }
+
+    current = current.parent;
+  }
+
+  return null;
+}
+
+/**
+ * Returns host children for given element.
+ * @param element The element start traversing from.
+ */
+export function getHostChildren(
+  element: ReactTestInstance | null
+): ReactTestInstance[] {
+  if (element == null) {
+    return [];
+  }
+
+  const hostChildren: ReactTestInstance[] = [];
+
+  element.children.forEach((child) => {
+    if (typeof child !== 'object') {
+      return;
+    }
+
+    if (isHostElement(child)) {
```

Comment:
If I understand correctly, if a child host element itself has host children, only this child host element will be returned. Is that on purpose?

Reply:
Yes, as this method returns host children, i.e. direct host descendants, not all host descendents